### ISSUE TITLE
Foreign key editor can not add correct default name if foreign keys if another one already exists

### ISSUE THREAD

**Issue by scifood** [phpmyadmin/phpmyadmin#12682]
### Steps to reproduce
1. Go to relation view
2. Make foreign relation with empty relation key name (new fk with name %tablename%_ibfk_1 will be created)
3. Make another foreign relation with empty relation key name

### Expected behaviour
New fk with name %tablename%_ibfk_2 will be created

### Actual behaviour
But %tablename%_ibfk_1 trying to create one more time

**Comment by devenbansod** [comment]
Hi, which version are you using ? 

I could not actually reproduce the problem. I added the first FK followed by the second one, and it gave the correct name for the second as well. 
Could you please try and reproduce it on our demo server at : https://demo.phpmyadmin.net/QA_4_6/?pma_username=root ?

**Comment by scifood** [comment]
Version information: 4.6.4
Bug able to repeated at your server. See DB "titch"
![screenshot from 2016-11-08 16-29-48](https://cloud.githubusercontent.com/assets/22497367/20100980/e825c94c-a5d0-11e6-8b6b-16ea864ca8a5.png)

**Comment by devenbansod** [comment]
Actually it seems to be a problem with MySQL itself. 
1. If I remove the `$` in the referencing table name (originally `test$_foregn`) and retry, it works perfectly well.
2. Also, we don't generate the constraint names in phpMyAdmin, rather they are automatically generated by the MySQL server if you don't provide in the constraint name.

**Comment by scifood** [comment]
No, it's not. If I set manually name for FK as test_foregn_ibfk_2 all work's perfectly.

**Comment by devenbansod** [comment]
I agree that it works perfectly fine in case of a constraint name being provided. 

But, are you able to run the query 
```
ALTER TABLE `test$_foregn` ADD  FOREIGN KEY (`faid`) REFERENCES `test$_`(`id`) ON DELETE RESTRICT ON UPDATE RESTRICT;
```
on the terminal or the SQL tab (assuming the foreign key from `fid` to `id` has already been added from the UI) ? This is the query which we are trying to run when adding a foreign key without a constraint name.

**Comment by scifood** [comment]
I have only used UI for both FK.

> This is the query which we are trying to run when adding a foreign key without a constraint name.
I understand that. Unfortunately, this SQL is absolutely correct. Maybe there is a way how to make the addition of a key name more convenient? Empty field - an ideal option, but it does not work as it should in this case. Could you make a subquery to get the names of the existing keys and fill the empty names similar to those used by default? This would solve the problem, but the button "Preview the SQL" will not operate correctly without fix.

**Comment by devenbansod** [comment]
If it helps the user experience, we could do it surely and add this an enhancement targeting v4.7.0. 

BTW I think the exact same thing works for the MariaDB (you can reproduce it on demo server's MariaDB version, just to make sure it's not a phpMyAdmin 'bug' after all), but as I said if it helps the user experience, I am adding this as an enhancement (targeted for next feature-release v4.7.0)

**Comment by scifood** [comment]
Many thnx!

**Comment by scifood** [comment]
@devenbansod, works on 4.7.0 already in progress but this issue is not applied to milestone. Does it mean that this feature will not be implemented in 4.7.0? Also this question may be duplicated for https://github.com/phpmyadmin/phpmyadmin/issues/12697
